Engine/Source/Timer.cpp

```diff
@@ -26,10 +26,11 @@ void Timer::Update()
 		mDeltaTime = ReadDelta(); 
 	}
 
-	//Delay the frame so the FPS match the limit if mDeltaTime 
-	if (mFpsLimit > 0 && mDeltaTime / mSpeed < (1000 / mFpsLimit))
+	//Delay the frame so the FPS match the limit if mDeltaTime if  vsync isn't enabled
+	if (mFpsLimit > 0 && mDeltaTime / mSpeed < (1000 / mFpsLimit) && !mEnabledVsync)
```

Comment:
In C++ the logical AND operator (&&) employs short-circuit evaluation. This means that if the left operand of the && operator evaluates to false, the right operand is not evaluated because the entire expression will always be false regardless of the right operand's value. 
For this case, I'd propose to put first the !mEnabledVsync condition, and then all the other conditional operators. Also, the other two operators (`mFpsLimit > 0 && mDeltaTime / mSpeed < (1000 / mFpsLimit`) seems that goes togethjer, so this would be my proposal: 
`if (!mEnabledVsync && (mFpsLimit > 0 && mDeltaTime / mSpeed < (1000 / mFpsLimit))`
What do you think?

Reply:
@MarcGabernet FYI